vunit/sim_if/modelsim.py

```diff
@@ -218,13 +218,17 @@ def create_library(self, library_name, path, mapped_libraries=None):
         cfg.set("Library", library_name, path)
         write_modelsimini(cfg, self._sim_cfg_file_name)
 
-    def _get_mapped_libraries(self):
+    def _get_mapped_libraries(self, cfg_file_name=None):
         """
         Get mapped libraries from modelsim.ini file
         """
-        cfg = parse_modelsimini(self._sim_cfg_file_name)
+        cfg_file_name = cfg_file_name if cfg_file_name is not None else self._sim_cfg_file_name
+        cfg = parse_modelsimini(cfg_file_name)
         libraries = dict(cfg.items("Library"))
         if "others" in libraries:
+            other_libraries = self._get_mapped_libraries(libraries["others"])
+            # current mappings take precedence over others
+            libraries = {**other_libraries, **libraries}
```

Comment:
Can this be a single line?

```py
libraries = {**(self._get_mapped_libraries(libraries["others"])), **libraries}
```

Reply:
Sure. I'd prefer two statements for readability, but I can change this if you like.

Reply:
It's ok. Just checking that `other_libraries` is not used somewhere else (it's local to the if cluse).